tests/components/yale/mocks.py

```diff
@@ -18,108 +19,192 @@
     DoorbellViewActivity,
     DoorOperationActivity,
     LockOperationActivity,
 )
-from yalexs.authenticator import AuthenticationState
+from yalexs.api_async import ApiAsync
+from yalexs.authenticator_common import Authentication, AuthenticationState
 from yalexs.const import Brand
 from yalexs.doorbell import Doorbell, DoorbellDetail
 from yalexs.lock import Lock, LockDetail
+from yalexs.manager.ratelimit import _RateLimitChecker
 from yalexs.pubnub_async import AugustPubNub
 
-from homeassistant.components.yale.const import CONF_BRAND, CONF_LOGIN_METHOD, DOMAIN
+from homeassistant.components.application_credentials import (
+    ClientCredential,
+    async_import_client_credential,
+)
+from homeassistant.components.yale.const import DOMAIN
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
 from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
 
 from tests.common import MockConfigEntry, load_fixture
 
+USER_ID = "a76c25e5-49aa-4c14-cd0c-48a6931e2081"
+
 
-def _mock_get_config(brand: Brand = Brand.YALE_HOME):
+def _mock_get_config(
+    brand: Brand = Brand.YALE_GLOBAL, jwt: str | None = None
+) -> dict[str, Any]:
     """Return a default yale config."""
     return {
         DOMAIN: {
-            CONF_LOGIN_METHOD: "email",
-            CONF_USERNAME: "mocked_username",
-            CONF_PASSWORD: "mocked_password",
-            CONF_BRAND: brand,
+            "auth_implementation": "yale",
+            "token": {
+                "access_token": jwt or "access_token",
+                "expires_in": 1,
+                "refresh_token": "refresh_token",
+                "expires_at": time.time() + 3600,
+                "service": "yale",
+            },
         }
     }
 
 
-def _mock_authenticator(auth_state):
+def _mock_authenticator(auth_state: AuthenticationState) -> Authentication:
     """Mock an yale authenticator."""
     authenticator = MagicMock()
     type(authenticator).state = PropertyMock(return_value=auth_state)
     return authenticator
 
 
-def _timetoken():
+def _timetoken() -> str:
     return str(time.time_ns())[:-2]
 
 
-@patch("yalexs.manager.gateway.ApiAsync")
-@patch("yalexs.manager.gateway.AuthenticatorAsync.async_authenticate")
-async def _mock_setup_yale(
-    hass, api_instance, pubnub_mock, authenticate_mock, api_mock, brand
-):
-    """Set up yale integration."""
-    authenticate_mock.side_effect = MagicMock(
-        return_value=_mock_yale_authentication(
-            "original_token", 1234, AuthenticationState.AUTHENTICATED
-        )
-    )
-    api_mock.return_value = api_instance
-    entry = MockConfigEntry(
+async def mock_yale_config_entry_and_client_credentials(
+    hass: HomeAssistant,
+) -> ConfigEntry:
+    """Mock yale config entry and client credentials."""
+    entry = mock_config_entry()
+    entry.add_to_hass(hass)
+    await mock_client_credentials(hass)
```

Comment:
Considered just adding autouse to the fixture?

Reply:
26e20135fbcfeaecaa29bc0e37c1fca407493b4e